dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/put-fee-record-correction-transient-form-data.controller/helpers/validate-record-correction-transient-form-values.ts

```diff
@@ -0,0 +1,156 @@
+      return formValues.utilisation;
+    case RECORD_CORRECTION_REASON.OTHER:
+      return formValues.additionalComments;
+    default:
+      throw new Error(`Invalid record correction reason: ${reason}`);
+  }
+};
+
+/**
+ * Validates that there are no form values provided for reasons that are not in the correction request.
+ *
+ * Note that the "additionalComments" field is required when {@link RECORD_CORRECTION_REASON.OTHER} is
+ * one of the reasons, and optional when {@link RECORD_CORRECTION_REASON.OTHER} is not one of the
+ * reasons. It is therefore not included in the "unexpected reasons" check as it can be provided
+ * regardless of the correction reasons.
+ * @param formValues - The form values to validate.
+ * @param reasons - The reasons for the record correction that determine which form values are expected.
+ * @throws Error if there are any form values provided for reasons not in the correction request.
+ */
+export const validateNoUnexpectedReasonValues = (formValues: RecordCorrectionFormValues, reasons: RecordCorrectionReason[]) => {
+  const unexpectedReasons = difference(Object.values(RECORD_CORRECTION_REASON), [...reasons, RECORD_CORRECTION_REASON.OTHER]);
+
+  unexpectedReasons.forEach((reason) => {
+    const reasonFormValue = getFormValueForReason(formValues, reason);
+
+    if (reasonFormValue) {
+      throw new Error(`Unexpected form value "${reasonFormValue}" for reason "${reason}" not in the correction request reasons.`);
+    }
+  });
+};
+
+/**
+ * Defines validation rules for each record correction reason, mapping each reason to a validation function that returns validation errors.
+ */
+type ValidatorMap = Record<
+  RecordCorrectionReason,
+  (value?: string) => Promise<Partial<RecordCorrectionFormValueValidationErrors>> | Partial<RecordCorrectionFormValueValidationErrors>
+>;
+
+/**
+ * Creates a map of validation functions for each record correction reason.
+ * @param reasons - Array of record correction reasons that determine validation rules
+ * @returns A ValidatorMap object containing validation functions for each record correction reason type
+ */
+const getValidatorMap = (reasons: RecordCorrectionReason[]): ValidatorMap => {
+  const facilityIdValidator = async (value?: string) => ({ facilityIdErrorMessage: await getFacilityIdValidationError(value) });
+  const reportedCurrencyValidator = (value?: string) => (!isCurrencyValid(value) ? { reportedCurrencyErrorMessage: 'You must select a currency' } : {});
+  const reportedFeeValidator = (value?: string) =>
+    !isMonetaryAmountValid(value) ? { reportedFeeErrorMessage: 'You must enter the reported fee in a valid format' } : {};
+  const utilisationValidator = (value?: string) =>
+    !isMonetaryAmountValid(value) ? { utilisationErrorMessage: 'You must enter the utilisation in a valid format' } : {};
+  const otherReasonValidator = (value?: string) => ({ additionalCommentsErrorMessage: getAdditionalCommentsValidationError(reasons, value) });
```

Comment:
Is it possible to write all of these in the shape:
```
const fieldValidator = async (value?: string) => ({ fieldErrorMessage: await getFieldValidationError(value) });
```
Then define `getFieldValidationError` separately? I think it would make it significantly easier to read

Also can we have new lines after function definitions please?